HousingManagementSystemApi/Gateways/PropertiesGateway.cs

```diff
@@ -0,0 +1,43 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Threading.Tasks;
+using HACT.Dtos;
+
+namespace HousingManagementSystemApi.Gateways
+{
+    using Ardalis.GuardClauses;
+    using Models;
+
+    public class PropertiesGateway : IAddressesGateway
+    {
+        private readonly IHttpClientFactory httpClientFactory;
+
+        public PropertiesGateway(IHttpClientFactory httpClientFactory)
+        {
+            this.httpClientFactory = httpClientFactory;
+        }
+
+        public async Task<IEnumerable<PropertyAddress>> SearchByPostcode(string postcode)
+        {
+            Guard.Against.NullOrWhiteSpace(postcode, nameof(postcode));
+
+            var httpClient = httpClientFactory.CreateClient(HttpClientNames.Properties);
+            var request = new HttpRequestMessage(HttpMethod.Get,
+                $"properties?postcode={postcode}&pageSize=100");
```

Comment:
Are we planning on handling responses that are over 100- if so we probably need a test for it too

Reply:
We have guidance that the approximate maximum will be 50 addresses within the response so at this time I feel using a page size of 100 would be fine.